cmd/kube-controller-manager/app/plugins.go

```diff
@@ -95,32 +97,18 @@ func ProbeRecyclableVolumePlugins(config componentconfig.VolumeConfiguration) []
 	}
 	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins(nfsConfig)...)
```

Comment:
That was too magical.  I meant in the other direction.  We could have a
pkg/volume/all/all_plugins.go which explicitly contains a list of all
plugins and offers simple functions like `all.ProbeVolumePlugins`.  This is
probably a different set of PRs to do this.  My main gripe is that
controller manager is special-casing things like NFs, when it really should
be generic mechanism.  Plugins should be more or less opaque.  I'll file an
issue

On Mon, Aug 8, 2016 at 4:05 AM, Jan Šafránek <notifications@github.com>
wrote:

> In cmd/kube-controller-manager/app/plugins.go
> <https://github.com/kubernetes/kubernetes/pull/29006#discussion_r73857700>
> :
>
> > @@ -95,32 +97,18 @@ func ProbeRecyclableVolumePlugins(config componentconfig.VolumeConfiguration) []
> >  	}
> >  	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins(nfsConfig)...)
>
> This approach was turned down in PR #21552
> <https://github.com/kubernetes/kubernetes/pull/21552>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29006/files/ec13593208a8c485eaded742cd20663864293c05#r73857700>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVETXYdApodtqcD-BBXfppNjNTXHAks5qdw1ygaJpZM4JNdmW>
> .
>